src/main/java/seedu/duke/logic/parser/ParserUtils.java

```diff
@@ -22,410 +21,278 @@
-                    params.split(Command.COMMAND_PREFIX_END_DATE
-                            + Command.COMMAND_PREFIX_DELIMITER, 2)[1];
-            String dateString = extractRelevantParameter(stringAfterPrefix);
-            logger.log(Level.WARNING, String.format("date string detected is: %s", dateString));
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(ParserMessages.DATE_FORMAT);
-            return LocalDate.parse(dateString, formatter);
-        } catch (IndexOutOfBoundsException e) {
-            logger.log(Level.WARNING, "Detected empty end date input after prefix but date is required!");
-            throw new ParamMissingException(ParserMessages.MESSAGE_ERROR_NO_END_DATE);
-        } catch (DateTimeParseException e) {
-            throw new ParamInvalidException(ParserMessages.MESSAGE_ERROR_INVALID_DATE_FORMAT);
+            return extractGeneralDouble(params, Command.COMMAND_PREFIX_WEIGHT);
+        } catch (InvalidParamException e) {
+            logger.log(Level.WARNING, "Detected non-digit weight input.");
+            throw new ParserException(String.format(ParserMessages.MESSAGE_ERROR_NOT_A_NUMBER, "weight"));
         }
     }
-    //@@author
 
     protected static LocalDate extractDate(String params, boolean isRequired)
-            throws ParamInvalidException, ParamMissingException {
-        try {
-            String stringAfterPrefix =
-                    params.split(Command.COMMAND_PREFIX_DATE
-                            + Command.COMMAND_PREFIX_DELIMITER, 2)[1];
-            String dateString = extractRelevantParameter(stringAfterPrefix);
-            logger.log(Level.WARNING, String.format("date string detected is: %s", dateString));
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(ParserMessages.DATE_FORMAT);
-            LocalDate date = LocalDate.parse(dateString, formatter);
-            YearMonth currentMonth = YearMonth.of(date.getYear(), date.getMonth());
-            if (Integer.parseInt(dateString.substring(0, dateString.indexOf("-")))
-                    > currentMonth.atEndOfMonth().getDayOfMonth()) {
-                throw new ParamInvalidException(ParserMessages.MESSAGE_ERROR_INVALID_DATE_FORMAT);
-            }
-            return date;
-        } catch (IndexOutOfBoundsException e) {
-            if (isRequired) {
-                logger.log(Level.WARNING, "Detected empty date input after prefix but date is required!");
-                throw new ParamMissingException(ParserMessages.MESSAGE_ERROR_NO_DATE);
-            } else {
-                logger.log(Level.WARNING, "Detected empty date input after prefix, assuming date to be now");
-                return LocalDate.now();
-            }
-        } catch (DateTimeParseException e) {
-            throw new ParamInvalidException(ParserMessages.MESSAGE_ERROR_INVALID_DATE_FORMAT);
+            throws ParserException {
+        LocalDate localDate = extractGeneralDate(params, Command.COMMAND_PREFIX_DATE);
```

Comment:
Just curious where is the implementation of the method `extractGeneralDate`, since I cannot find it :(

Reply:
It starts from line 85 of `ParserUtils`! :)

Reply:
Thank you! It looks good!